### PR TITLE
Update notebooks/official/matching_engine/README.md

### PR DESCRIPTION
**Pull request by jsimonweb**
Update notebooks/official/matching_engine/README.md :

- Add Title heading with list of notebooks
- Add code snippet to each notebook's details
- Add href anchor id's for direct linking to a specific notebook's details in README

### CODE REVIEW

File: notebooks/official/matching_engine/README.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
+## Vertex-AI: Matching Engine Notebooks
+  - [Create Vertex AI Matching Engine index](#sdk_matching_engine_for_indexing)
+  - [Introduction to builtin Swivel embedding algorithm](#intro-swivel)
+  - [Introduction to builtin Two-towers embedding algorithm](#two-tower-model-introduction)
+---
+---
 
-[Create Vertex AI Matching Engine index](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb)
+
+<a id="sdk_matching_engine_for_indexing"></a>[Create Vertex AI Matching Engine index](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb)
```
Comment (ivanmkc) [inline notebooks/official/matching_engine/README.md:9]: I assume this is like a bookmark.

File: notebooks/official/matching_engine/README.md [hunk 2]
```diff
@@ -11,8 +18,27 @@ The steps performed include:
 * Perform online query
 * Compute recall
 
+    <details>
```
Comment (ivanmkc) [inline notebooks/official/matching_engine/README.md:21]: Is this a markdown tag?

**Review by ivanmkc — APPROVED** [review]
LGTM

Left note in email.

File: notebooks/official/matching_engine/README.md (hunk 1 above)
Comment (jsimonweb) [reply notebooks/official/matching_engine/README.md:9]: Correct.

File: notebooks/official/matching_engine/README.md (hunk 2 above)
Comment (jsimonweb) [reply notebooks/official/matching_engine/README.md:21]: In github markdown this creates an expandable/collapsible element:

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

**Pull request merged**